libs/src/services/solanaWeb3Manager/rewards.js

```diff
@@ -0,0 +1,767 @@
+    },
+    {
+      pubkey: transferAccount,
+      isSigner: false,
+      isWritable: true,
+    },
+    {
+      pubkey: derivedAAOAddress,
+      isSigner: false,
+      isWritable: false,
+    },
+    {
+      pubkey: feePayer,
+      isSigner: true,
+      isWritable: true,
+    },
+    {
+      pubkey: SYSVAR_RENT_PUBKEY,
+      isSigner: false,
+      isWritable: false,
+    },
+    {
+      pubkey: TOKEN_PROGRAM_ID,
+      isSigner: false,
+      isWritable: false,
+    },
+    {
+      pubkey: SystemProgram.programId,
+      isSigner: false,
+      isWritable: false,
+    },
+  ]
+
+  // Construct the instruction data
+  const instructionData = new ValidateAttestationsInstructionData({
+    amount: tokenAmount.toNumber(),
+    id: transferId,
+    eth_recipient: ethAddressToArray(recipientEthAddress),
+  })
+  const serializedInstructionData = borsh.serialize(
+    validateAttestationsInstructionSchema,
+    instructionData
+  )
+  const serializedInstructionEnum = Buffer.from(
+    Uint8Array.of(EVALUATE_INSTRUCTION_ENUM_VALUE, ...serializedInstructionData)
+  )
+  const transferInstruction = new TransactionInstruction({
+    keys: accounts,
+    programId: rewardManagerProgramId,
+    data: serializedInstructionEnum,
+  })
+
+  // Prepare and send transaction
+  const relayable = prepareInstructionForRelay(transferInstruction)
+  const { blockhash: recentBlockhash } = await connection.getRecentBlockhash()
+  const transactionData = {
+    recentBlockhash,
+    instructions: relayable,
+  }
+
+  try {
+    const response = await identityService.solanaRelay(transactionData)
+    return response
+  } catch (e) {
+    console.error(e.message)
+    console.log({ e })
+  }
+}
+
+// Helpers
+
+// Generate particular instructions
+
+/**
+ *
+ * Helper function to generate a submit attestation instruction.
+ * @param {{
+ *   attestationMeta: AttestationMeta,
+ *   derivedMessageAccount: PublicKey
+ *   rewardManagerAccount: PublicKey
+ *   rewardManagerAuthority: PublicKey
+ *   rewardManagerProgramId: PublicKey
+ *   feePayer: PublicKey
+ *   transferId: string
+ * }} {
+ *   attestationMeta,
+ *   derivedMessageAccount,
+ *   rewardManagerAccount,
+ *   rewardManagerAuthority,
+ *   rewardManagerProgramId,
+ *   feePayer,
+ *   transferId
+ * }
+ * @returns {Promise<TransactionInstruction>}
+ */
+const generateSubmitAttestationInstruction = async ({
+  attestationMeta,
+  derivedMessageAccount,
+  rewardManagerAccount,
+  rewardManagerAuthority,
+  rewardManagerProgramId,
+  feePayer,
+  transferId,
+}) => {
+  // Get the DN's derived Solana address from the eth pubkey
+  const derivedSender = await deriveSolanaSenderFromEthAddress(
+    attestationMeta.ethAddress,
+    rewardManagerProgramId,
+    rewardManagerAccount
+  )
+
+  ///   Verify transfer signature
+  ///
+  ///   0. `[writable]` New or existing account storing verified messages
+  ///   1. `[]` Reward manager
+  ///   1. `[]` Reward manager authority (NEW)
```

Comment:
NEW?

Reply:
lol notes to myself, thanks for catching